vendor/bat-native-ledger/src/bat/ledger/internal/state/state_migration_v4.cc

```diff
@@ -0,0 +1,21 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "bat/ledger/internal/state/state_migration_v4.h"
+
+#include "bat/ledger/internal/ledger_impl.h"
+
+namespace braveledger_state {
+
+StateMigrationV4::StateMigrationV4(bat_ledger::LedgerImpl* ledger) :
+    ledger_(ledger) {
+}
+
+StateMigrationV4::~StateMigrationV4() = default;
+
+void StateMigrationV4::Migrate(ledger::ResultCallback callback) {
+  ledger_->ledger_client()->DeleteLogs(callback);
```

Comment:
I assume `LedgerImpl::OnStateInitialized` and the callback do not create a new diagnostic log as they call `BLOG`?

Reply:
it can not as we have flag in rewards_service. So no matter when we call this no new logs will be logged